daemon/qrexec-daemon-common.c

```diff
@@ -84,7 +84,7 @@ int handle_daemon_handshake(int fd)
         return -1;
     }
 
-    if (info.version != QREXEC_PROTOCOL_VERSION) {
+    if (info.version < QREXEC_PROTOCOL_V3) {
```

Comment:
Generally, it's not enough to just relax this check. You need to also save which version the daemon is using (V4 is just adding a message that isn't relevant for the qrexec-client, but still). And similarly change in `handle_client_hello` and also save which client is using what version.

Alternatively, you can keep using V3 for daemon<->client communication, since new message is not needed at this place (right?). This will be also easier to test. But relaxing the check is a good idea for the future anyway (but can be done in a separate PR if you prefer).